### PR TITLE
Is filt

### PR DESCRIPTION
**Pull request by Redmar-van-den-Berg**
I have implemented the convenience property is_filt for both _Record and _Call, to indicate whether a variant or single call has been filtered. 

I often work with filtered vcf files, and so I found myself writing a bit of code to check whether a call or variant had been filtered. Like is_snp or is_indel, I figured is_filt would be useful to have in PyVCF itself.

### CODE REVIEW

File: vcf/model.py [hunk 1]
```diff
@@ -117,6 +117,22 @@ def is_het(self):
             return None
         return self.gt_type == 1
 
+    @property
+    def is_filt(self):
```
Comment (martijnvermaat) [inline vcf/model.py]: I don't think we have to abbreviate this and feel that `is_filtered` might be a better name, what do you think?

File: vcf/model.py [hunk 2]
```diff
@@ -117,6 +117,22 @@ def is_het(self):
...
+        """ Return True for filtered calls """
+        try: # no FT annotation present for this variant
+            FT=self.data.FT
```
Comment (martijnvermaat) [inline vcf/model.py]: Could you [PEP 8](https://www.python.org/dev/peps/pep-0008/)-ify this? Preferably no uppercase variable names and spacing around the `=`.

File: vcf/model.py [hunk 3]
```diff
@@ -117,6 +117,22 @@ def is_het(self):
...
+        except AttributeError:
+            return False
+        if FT == None or FT == []: # FT is not set or set to PASS
```
Comment (martijnvermaat) [inline vcf/model.py]: Compare to `None` with `is` (not `==`) and check for the empty list with `len(FT) == 0`.

File: vcf/model.py [hunk 4]
```diff
@@ -117,6 +117,22 @@ def is_het(self):
...
+            return False
+        elif len(FT) > 0: # FT contains one or more filters
```
Comment (martijnvermaat) [inline vcf/model.py]: Shouldn't this case already be taken care of by the one above?

File: vcf/model.py [hunk 5]
```diff
@@ -117,6 +117,22 @@ def is_het(self):
...
+            return True
+        else: # This should not happen
```
Comment (martijnvermaat) [inline vcf/model.py]: I don't think it is needed to add special casing for things that can never happen. I'd just have a default `return True` or `return False` as the default case.

**Comment by martijnvermaat** [comment]
Thanks! I have added a few stylistic liner notes. In general agree with the utility of this.

**Comment by Redmar-van-den-Berg** [comment]
Thanks for your feedback. I agree that is_filtered is a better name, so I have modified this for both _Record and _Call. I have also made the other changes you suggested (PEP-8, and I've removed the trailing else clauses)

**Comment by martijnvermaat** [comment]
Thank you!

**Pull request merged**